Metric_Publisher/Functions/Metrics.py

```diff
@@ -0,0 +1,56 @@
+from Metric_Publisher.Functions.Metric import Metric
+from contextlib import contextmanager
+import time
+
+
+class TimingMetric(Metric):
+    """Timing metric that calculates the time the metric has run."""
+
+    def __init__(self, metric_name: str, unit: str):
+        super().__init__(metric_name, unit)
+
+    @contextmanager
+    def timeit_metric(self):
+        """Context manager used to calculate the time the method has run."""
+        start_time = 0
+        try:
+            start_time = time.time()
+            yield
+        finally:
+            self._value = time.time() - start_time
+
+    def get_name(self) -> str:
+        """Returns the metric name."""
+        return super().get_name()
+
+    def get_unit(self) -> str:
+        """Returns the unit associated with the metric."""
+        return super().get_unit()
+
+    def get_value(self):
+        """Returns the value associated with the metric."""
+        return super().get_value()
+
+
+class CounterMetric(Metric):
+    """Counter metric that counts the key count."""
+
+    def __init__(self, metric_name: str, unit: str):
+        super().__init__(metric_name, unit)
+        self._value = 0
+
+    def inc_count_metric(self):
+        """Increments the count metric by 1."""
+        self._value += 1
```

Comment:
If this metric is going to be theoretically updated by many different servers at once, this will run into an issue:

Server A increment
A increment
A increment
Server A pushes a count of 3
Server B increment
Server B pushes a count of 1

This has B's value overriding the one that A wrote. Instead, I think you need some way for increment to grab the last value off of wherever it is and then immediately publish the next number. Unless is the expectation that one metric is being updated by one server? @thehcma please correct me if I'm wrong.

Reply:
If server A and Server B are not listed under the same instance Id (They shouldn't logically) wouldn't this be the desired result?

Suppose server A is generating 100 keys per hour, and Server B is generating 1 Key and hour, wouldn't we want to isolate each server to its metric? That way we can filter using the instance id . Just my thought process.

Reply:
Don't think so - our service can be used by many servers at once. Each server has its own folder in the bucket. Assuming each organization has a bucket with potentially multiple servers, we need to ensure that the count is relatively consistent across multiple servers.